### PR TITLE
AG-12025 Show active state for toolbar zoom

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-12025

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/zoom/zoomToolbar.ts [hunk 1]
```diff
@@ -78,6 +89,30 @@ export class ZoomToolbar {
...
+            const { selectedZoom } = this;
+            const { x } = e;
+
+            this.toolbarManager.toggleGroup('zoom-toolbar', 'ranges', { active: false });
+
+            if (
+                selectedZoom != null &&
+                x != null &&
+                Math.abs(selectedZoom.min - x.min) < 1e-9 &&
+                Math.abs(selectedZoom.max - x.max) < 1e-9
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/zoom/zoomToolbar.ts]: nit: Could we change this to `1e-10` to match `isZoomEqual()` in `zoomUtils.ts`?

I've had bugs with zooms not matching and refining the granularity of the floating point of zoom to be the same everywhere.

File: packages/ag-charts-community/src/chart/interaction/toolbarManager.ts [hunk 2]
```diff
@@ -44,19 +45,20 @@ export interface ToolbarFloatingAnchorChangedEvent extends Event<'floating-ancho
...
 
 export interface ToolbarButtonPressedEvent<T = any> extends Event<'button-pressed'> {
+    id: string;
     value: T;
     rect: BBox;
 }
 
-export interface ToolbarButtonToggledEvent<T = any> extends Event<'button-toggled'> {
-    value: T;
+export interface ToolbarButtonToggledEvent extends Event<'button-toggled'> {
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/interaction/toolbarManager.ts]: Can we keep this `value: T` typing? It is used in annotations https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-enterprise/src/features/annotations/annotations.ts#L358

**Review by lsjroberts — APPROVED** [review]
Nice! LGTM

**Review by alantreadway — APPROVED** [review]

File: packages/ag-charts-community/src/chart/interaction/toolbarManager.ts (hunk 2 above)
Comment (lsjroberts) [reply packages/ag-charts-community/src/chart/interaction/toolbarManager.ts]: ignore this, sorry

**Pull request merged**